tachyon/zk/lookup/logup/verifier.h

```diff
@@ -0,0 +1,133 @@
+// Copyright 2020-2022 The Electric Coin Company
+// Copyright 2022 The Halo2 developers
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.halo2 and the LICENCE-APACHE.halo2
+// file.
+
+#ifndef TACHYON_ZK_LOOKUP_LOGUP_VERIFIER_H_
+#define TACHYON_ZK_LOOKUP_LOGUP_VERIFIER_H_
+
+#include <memory>
+#include <vector>
+
+#include "tachyon/crypto/commitments/polynomial_openings.h"
+#include "tachyon/zk/lookup/halo2/opening_point_set.h"
+#include "tachyon/zk/lookup/logup/verifier_data.h"
+#include "tachyon/zk/lookup/lookup_argument.h"
+#include "tachyon/zk/lookup/verifier.h"
+#include "tachyon/zk/lookup/verifying_evaluator.h"
+#include "tachyon/zk/plonk/base/l_values.h"
+#include "tachyon/zk/plonk/halo2/proof.h"
+
+namespace tachyon::zk::lookup {
+
+namespace logup {
+
+template <typename F, typename C>
+class Verifier final : public lookup::Verifier<typename logup::Verifier<F, C>> {
+ public:
+  using Proof = plonk::halo2::LogUpProof<F, C>;
+
+  Verifier(const Proof& proof, size_t circuit_idx)
+      : data_(proof.ToLookupVerifierData(circuit_idx)) {}
+
+  Verifier(const Proof& proof, size_t circuit_idx,
+           const plonk::LValues<F>& l_values)
+      : data_(proof.ToLookupVerifierData(circuit_idx)), l_values_(&l_values) {}
+
+  void DoEvaluate(const std::vector<lookup::Argument<F>>& arguments,
+                  std::vector<F>& evals) {
+    lookup::VerifyingEvaluator<F> evaluator(data_);
+
+    F active_rows = F::One() - (l_values_->last + l_values_->blind);
```

Comment:
Is `F::One()` correct?
I see `C::Scalar::ZERO` [here](https://github.com/scroll-tech/halo2/blob/1070391642dd64b2d68b47ec246cba9e35bd3c15/halo2_proofs/src/plonk/mv_lookup/verifier.rs#L95).